src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```diff
@@ -355,92 +349,30 @@ public String getPlanKey(Object requestBody) throws Exception {
 
     @Override
     public Result onBuildCompleted(ProgrammingExerciseParticipation participation, Object requestBody) {
-        final var report = TestResultsDTO.convert(requestBody);
-        final var latestPendingSubmission = programmingSubmissionRepository.findByParticipationIdAndResultsIsNullOrderBySubmissionDateDesc(participation.getId()).stream()
-                .filter(submission -> {
-                    final var commitHash = getCommitHash(report, submission.getType());
-                    return commitHash.isPresent() && submission.getCommitHash().equals(commitHash.get());
-                }).findFirst();
-        var result = createResultFromBuildResult(report, (Participation) participation);
-        final ProgrammingSubmission submission;
-        submission = latestPendingSubmission.orElseGet(() -> createFallbackSubmission(participation, report));
-        submission.setBuildFailed(result.getResultString().equals("No tests found"));
+        final var buildResult = TestResultsDTO.convert(requestBody);
+        var newResult = createResultFromBuildResult(buildResult, participation);
+
+        // Fetch submission or create a fallback
+        var latestSubmission = super.getSubmissionForBuildResult(participation.getId(), buildResult).orElseGet(() -> createAndSaveFallbackSubmission(participation, buildResult));
+        latestSubmission.setBuildFailed(newResult.getResultString().equals("No tests found"));
```

Comment:
we should set the build to failed, if the result was not successful to show the corresponding build log. So we should not check for "No tests found" but instead for newResult.isSuccessful()